File: fixtures/python/infinite_loop_no_exit/negative.py
# MAINT010-infinite-loop-no-exit: negative fixture
# These loops should NOT be flagged.

def loop_with_break(x):
    while True:
        if x:
            break

def loop_with_return():
    while True:
        return None

def loop_not_true(x):
    while x > 0:
        x -= 1

File: fixtures/rust/infinite_loop_no_exit/negative.rs
// MAINT010-infinite-loop-no-exit: negative fixture
// These loops should NOT be flagged.

fn loop_with_break(x: bool) {
    loop {
        if x {
            break;
        }
    }
}

fn loop_with_return() {
    loop {
        return;
    }
}

fn loop_with_panic() {
    loop {
        panic!("unexpected");
    }
}
